src/explore-education-statistics-common/src/modules/find-statistics/hooks/useTableQuery.ts

```diff
@@ -32,18 +40,28 @@ export default function useTableQuery(
 
     const queryKey = JSON.stringify(query);
 
-    let response = await storageService
-      .get<TableDataResponse>(queryKey)
+    let item = await storageService
+      .get<TableDataStorageItem>(queryKey)
       .catch(() => null);
 
-    if (!response) {
-      response = await tableBuilderService.getTableData(query, releaseId);
+    const dataExpired =
+      item != null &&
+      dataLastPublished !== '' &&
+      new Date(dataLastPublished) > new Date(item.dataLastPublished);
+
+    if (item == null || dataExpired) {
```

Comment:
Think this proposed invalidation technique is a bit overkill. If i'm not mistaken, all we have to do is add `dataLastPublished` into the serialized `queryKey` and job done.

```typescript
const queryKey = JSON.stringify({
  ...query,
  dataLastPublished,
});
```

Reply:
good idea sir

Reply:
Done - Thanks